metafacture-flux/src/main/antlr/org/metafacture/flux/parser/Flux.g

```diff
@@ -33,16 +33,53 @@ tokens {
 
 @header {
 package org.metafacture.flux.parser;
+
+import org.metafacture.flux.FluxParseException;
 }
 
 @lexer::header {
 package org.metafacture.flux.parser;
 }
 
+@parser::members {
+    // ensure throwing an exception
+    @Override
+    protected Object recoverFromMismatchedToken(IntStream input, int ttype, BitSet follow) throws RecognitionException {
+      {
+        RecognitionException e = new MismatchedTokenException(ttype, input);
+        // if next token is what we are looking for then "delete" this token
+        if ( mismatchIsUnwantedToken(input, ttype) ) {
+          e = new UnwantedTokenException(ttype, input);
+          beginResync();
+          input.consume(); // simply delete extra token
+          endResync();
+          reportError(e);  // report after consuming so AW sees the token in the exception
+          // we want to return the token we're actually matching
+          Object matchedSymbol = getCurrentInputSymbol(input);
+          input.consume(); // move past ttype token as if all were ok
```

Comment:
2 x `input.consume()`: Doesn't this consume two tokens?

Reply:
Yeah, good question ... to be honest, I did not totally traced this, just copied the code (as it was before, from the generated class).

Reply:
Ah, okay.